lib/src/Base/Stat/IsotropicCovarianceModel.cxx

```diff
@@ -146,16 +146,18 @@ Description IsotropicCovarianceModel::getFullParameterDescription() const
 /** Scale accessor */
 void IsotropicCovarianceModel::setScale(const Point & scale)
 {
-  if (scale.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the scale should have dimension 1, not " << scale.getDimension() << ".";
-  if (!(scale[0] > 0))
-    throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setScale";
+  if (scale.getDimension() != 1) throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setScale: the scale should have dimension 1, not " << scale.getDimension() << ".";
+  if (!(scale[0] > 0.0))
+    throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setScale: the scale is << " << scale[0] << " but should be positive";
   kernel_.setScale(scale);
   scale_ = scale;
 }
 
 /* Nugget factor accessor */
 void IsotropicCovarianceModel::setNuggetFactor(const Scalar nuggetFactor)
 {
+  if (!(nuggetFactor >= 0.0))
```

Comment:
I think this check is already done by `kernel_.setNuggetFactor`

Reply:
True, but doing it here allows to catch the exception earlier and produce an error message which might be a little more clear. However, if you think this is overkill, I can remove it.